docs/source/DeprecationPage.md

```diff
@@ -1,25 +1,166 @@
-  | Embedded types (`argument_type`, `first_argument_type`, and `second_argument_type`) in `std::function` | Use `decltype` if required | Deprecation in `Kokkos_Functional.hpp` mirrors that in `std::function` (`#include <functional>`) 
-  | `InitArguments` struct | `InitializationSettings()` object with attributes that can be queried | Make initialization transparent and understandable
-  | `finalize_all()` | `finalize()`| Improve  API
-  | Command-line arguments (other than `--help`) that are not prefixed with `kokkos-*` | `--kokkos-num-threads`, `--kokkos-device-id`, `--kokkos-num-devices`| Improve API
-  | `--[kokkos-]numa` command-line argument and `KOKKOS_NUMA` environment variable | `--kokkos-num-threads`| Harmonize option nomenclature with that of C++ `std::thread` library
-  | `--[kokkos-]threads` command-line argument | `--kokkos-num-threads` | Improve API
-  | Warn about `parallel_reduce` cases that call `join()` with arguments qualified by `volatile` keyword | Remove `volatile` overloads | Streamline API
-
-
-
-* Kokkos-4.0 will require C++17  
+  |  **Deprecated**  |  **Replacement**  |  **Reason**                
+  |  ------------  | ------------  |  ------------
+  |  `Kokkos::is_reducer_type` |  `Kokkos::is_reducer`  |  Improve API
+  |  Array reductions with raw pointer  |  Use `Kokkos::View` as return argument  |  Improve API
+  |  `OffsetView` constructors taking `index_list_type`  |  `Kokkos::pair` (CPU and GPU)  |  Streamline arguments to `::pair` function
+  |  Overloads of `Kokkos::sort` taking a parameter `bool always_use_kokkos_sort`  |  Use `Kokkos::BinSort` if required, or call `Kokkos::sort` without bool parameter  |  Updating overloads
+  |  |  **PUBLIC HEADERS UPDATES** 
+  |  Guard against non-public header inclusion  |  **Core PUBLIC HEADERS**:  |  Improve API
+  |  `Kokkos_Core.hpp`,
+  |  `Kokkos_Macros.hpp`,
+  |  `Kokkos_Atomic.hpp`,
+  |  `Kokkos_DetectionIdiom.hpp`,
+  |  `Kokkos_MathematicalConstants.hpp`,
+  |  `Kokkos_MathematicalFunctions.hpp`,
+  |  `Kokkos_NumericTraits.hpp`,
+  |  `Kokkos_Array.hpp`,
+  |  `Kokkos_Complex.hpp`,
+  |  `Kokkos_Pair.hpp`,
+  |  `Kokkos_Half.hpp`,
+  |  `Kokkos_Timer.hpp`
+  |  Guard against non-public header inclusion  |  **Algorithms PUBLIC HEADERS**:  |  Improve API
+  |  |  `Kokkos_StdAlgorithms.hpp`
+  |  |  `Kokkos_Random.hpp`,
+  |  |  `Kokkos_Sort.hpp`
+  |  Guard against non-public header inclusion  |  **Containers PUBLIC HEADERS**:  | Improve API
+  |  |  `Kokkos_Bitset.hpp`,
+  |  |  `Kokkos_DualView.hpp`,
+  |  |  `Kokkos_DynRankView.hpp`,
+  |  |  `Kokkos_ErrorReporter.hpp`,
+  |  |  `Kokkos_Functional.hpp`,
+  |  |  `Kokkos_OffsetView.hpp`,
+  |  |  `Kokkos_ScatterView.hpp`,
+  |  |  `Kokkos_StaticCrsGraph.hpp`,
+  |  |  `Kokkos_UnorderedMap.hpp`,
+  |  |  `Kokkos_Vector.hpp`
+  |  `Kokkos_UniqueToken.hpp` not a public header  |  `Kokkos_Core.hpp`  |  Improve API
```

Comment:
I found the following diffs in your list of private header files. I think this is because we renamed some files during the 3.7 release cycles. This is current as of release tag 3.7.01:

Missing:
```
Kokkos_AcquireUniqueTokenImpl.hpp
Kokkos_CopyViews.hpp
Kokkos_HostSpace.hpp
Kokkos_HPX.hpp
Kokkos_OpenMP.hpp
Kokkos_Parallel.hpp
Kokkos_WorkGraphPolicy.hpp
```

Extraneous (doesn't exist anymore, please remove):
```
Kokkos_WorkGraph.hpp
```

Reply:
For reference, here is what I used to obtain the list:

```bash
git grep --name-only 'KOKKOS_IMPL_WARNING("Including non-public Kokkos header files is not allowed.")' | awk -F '/' '{ print $NF; }' | sort
```